### PR TITLE
Esp folder layout

### PR DESCRIPTION
**Pull request by AlaiveCYA**

### CODE REVIEW

File: .github/workflows/esp_app_ci.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        uses: actions/checkout@v4
+        with:
+          submodules: 'recursive'
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.x'
+      - name: Install clang-format and  clang-tidy
+        run: sudo apt-get -y install clang-format clang-tidy
+
```
Comment (MaciejBinczarowski) [inline .github/workflows/esp_app_ci.yml:27]: I wonder if there is single command for format check

File: .github/workflows/esp_app_ci.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      - name: clang-format check
+        run: |
+            find . -name "*.c" -o -name "*.h" | xargs clang-format -i -style=file
+            git diff --exit-code
+
+      - name : Install ESP-IDF
+        run: |
+            cd ~
+            sudo apt-get install -y git wget flex bison gperf python3 python3-pip python3-venv cmake ninja-build ccache libffi-dev libssl-dev dfu-util libusb-1.0-0
```
Comment (MaciejBinczarowski) [inline .github/workflows/esp_app_ci.yml:36]: great way to do that 😅

File: .github/workflows/esp_app_ci.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            mkdir -p ~/esp_install
+            cd ~/esp_install
+            git clone -b v5.2.3 --recursive https://github.com/espressif/esp-idf.git
+            cd ~/esp_install/esp-idf
+            ./install.sh esp32
+      - name: Build and clang-tidy check
```
Comment (MaciejBinczarowski) [inline .github/workflows/esp_app_ci.yml:42]: and clang tidy check?

**Review by MaciejBinczarowski — APPROVED** [review]

File: .github/workflows/esp_app_ci.yml (hunk 3 above)
Comment (AlaiveCYA) [reply .github/workflows/esp_app_ci.yml:42]: It's not existing because I couldn't make it work

**Pull request merged**